pydruid/utils/aggregators.py

```diff
@@ -34,9 +37,26 @@ def max(raw_metric):
 def count(raw_metric):
     return {"type": "count", "fieldName": raw_metric}
 
+
 def hyperunique(raw_metric):
     return {"type": "hyperUnique", "fieldName": raw_metric}
 
+
+def filtered(filter, agg_input):
+    return {"type": "filtered",
+            "filter": Filter.build_filter(filter),
+            "aggregator": agg_input}
+
+
 def build_aggregators(agg_input):
-    return [dict([('name', k)] + list(v.items()))
-            for (k, v) in iteritems(agg_input)]
+    return [build_aggregator(name, kwargs)
+            for (name, kwargs) in iteritems(agg_input)]
+
+
+def build_aggregator(name, kwargs):
+    if kwargs["type"] == "filtered":
+        kwargs["aggregator"]["name"] = name
+    else:
+        kwargs.update({"name": name})
```

Comment:
Not sure if this affects anything, but before, if kwarks already contained a name it would use that instead of overriding it, since we passed the name as the first key/value pair to dict(...)